examples/step-27/step-27.cc

```diff
@@ -164,6 +172,29 @@ namespace Step27
   // face quadrature objects. We start with quadratic elements, and each
   // quadrature formula is chosen so that it is appropriate for the matching
   // finite element in the hp::FECollection object.
+  //
+  // Finally, we initialize FESeries::Fourier object which will be used to
+  // calculate coefficient in Fourier series as described in the introduction.
+  // In addition to the hp::FECollection, we need to provide quadrature rules
+  // hp::QCollection for integration on the reference cell.
+  //
+  // In order to resize fourier_coefficients Table, we use the following two
+  // auxiliary functions
+  template <typename T>
+  void resize(Table<2,T> &coeff, const unsigned int N)
+  {
+    coeff.reinit(N,N);
+  }
+
+  template <typename T>
+  void resize(Table<3,T> &coeff, const unsigned int N)
+  {
+    TableIndices<3> size;
+    for (unsigned int d=0; d<3; d++)
+      size[d] = N;
+    coeff.reinit(size);
+  }
```

Comment:
Can't you use the templated `TableIndices<dim>` class to cover all cases at once? Then you can avoid using these specializations...